### ISSUE TITLE
"Use of uninitialized value $pub in concatenation (.) or string at lib/CAP/Model/Parl.pm line 227."

### ISSUE THREAD

**Issue by RussellMcOrmond** [crkn-rcdr/cap#99]
The error in the title is very common in the Docker logs.  Likely input from outside (GET or POST request) wasn't sanitised before being used.

The same type of work should be done to resolve #14, where input isn't sanitised.

Also:

`Use of uninitialized value $host in pattern match (m//) at local/lib/perl5/Plack/Middleware/ReverseProxy.pm line 43.`

There are also attempts at attacks that are trying to use odd unicode characters to display/etc.

`[error] Caught exception in engine "utf8 "\xC0\xAF" does not map to Unicode at local/lib/perl5/Catalyst.pm line 3690."`

`Invalid \0 character in pathname for stat: /opt/cap/root/base/blocks/1some_inexistent_file_with_long_name\0.jpg/content-statement.html at local/lib/perl5/x86_64-linux-gnu/Template/Provider.pm line 971.`

In general we need to sanitise any input that is coming from outside, and never assume that only "valid" things will be requested.

**Comment by RussellMcOrmond** [comment]
This is apparently a duplicate of #73

**Comment by RussellMcOrmond** [comment]
A regular morning.  You can see with the errors that there are language errors, and people trying to do scripting test for cross-site scripting.

In many of these it is the language variable that comes from the browser isn't tested for validity before assuming it can be used to look for templates/etc.

```
cap-cap-1  | 2022/08/24-14:44:53 Starman::Server (type Net::Server::PreFork) starting! pid(1)
cap-cap-1  | Resolved [*]:3011 to [0.0.0.0]:3011, IPv4
cap-cap-1  | Binding to TCP port 3011 on host 0.0.0.0 with IPv4
cap-cap-1  | Setting gid to "1115 1115"
cap-cap-1  | [error] Couldn't render template "file error - blocks/ru/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/ru/content-statement.html: not found"
cap-cap-1  | [error] Caught exception in CAP::Controller::Browse->leaf "Use of uninitialized value $pub in concatenation (.) or string at lib/CAP/Model/Parl.pm line 227."
cap-cap-1  | [error] Couldn't render template "file error - blocks/fr'[0]/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/fr'[0]/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/"><script >alert(String.fromCharCode(88,83,83))</script>/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/"><script >alert(String.fromCharCode(88,83,83))</script>/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks//etc/passwd/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks//etc/passwd/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/Http://Www.Google.Com/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/Http://Www.Google.Com/blurb.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"
cap-cap-1  | [error] Couldn't render template "file error - blocks/da/content-statement.html: not found"

```

**Comment by RussellMcOrmond** [comment]
Duplicate of #73
